docs/implementing-new-trigger.md

```diff
@@ -0,0 +1,71 @@
+# How to add a new event source as Trigger
+
+Kubeless [architecture](/docs/architecture.md) is built on core concepts of Functions, Triggers and Runtime. A _Trigger_ in Kubeless represents association between an event source and functions that need to be invoked on a event in the event source. Kubeless fully leverages the Kubernetes concpets of [custom resource definition](https://kubernetes.io/docs/concepts/api-extension/custom-resources/)(CRD) and [custom controllers](https://kubernetes.io/docs/concepts/api-extension/custom-resources/#custom-controllers). Each trigger is expected to be modelled as Kubernetes CRD. A trigger specific custom resource controller is expected to be written that realizes how deployed functions are invoked when event occurs. Following sections document how one can add a new event source as _Trigger_ into Kubeless.
```

Comment:
`[architecture](/docs/architecture.md)` -> `[architecture](/docs/architecture)` (the final link won't be a md file but an html)
`on a event` -> `on an event`
`concpets` -> `concepts`